### PR TITLE
feat(er-next): rework navigation menu

### PR DESCRIPTION
**Pull request by DrShpongle**
This PR implements the following:

- `Modules` menu item is now visible for non-logged / non-registered users, they can navigate to the `/learn` page now.
- Logged in pro users get redirected from `/` to `/learn`, non-logged in users stay on `/` landing page
- Logged in users once they try to naviagate to `/login` get redirected to `/learn`
- Mobile menu gets closed after user clicks on any item in it

<details>
  <summary>**Modules** menu item for non-logged in users:</summary>
  
</details>
<details>
  <summary>Mobile menu gets closed after user clicks on any item in it</summary>
  
</details>

i guess d d GIF by Hyper RPG

### CODE REVIEW

File: apps/epic-react/src/components/app/navigation/mobile-navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+const MobileNavigation: React.FC<
+  React.PropsWithChildren<MobileNavigationProps>
+> = ({setOpen, children}) => {
+  const menuHolderRef = React.useRef<HTMLDivElement>(null)
+
+  const clickHandler = React.useCallback(
+    (e: MouseEvent) => {
+      const clickedElement = e.target as HTMLElement
+      const naviItem = clickedElement.closest('[data-navi-item]') as HTMLElement
```
Comment (DrShpongle) [inline apps/epic-react/src/components/app/navigation/mobile-navigation.tsx:15]: used event delegation approach here

File: apps/epic-react/src/components/app/navigation/index.tsx [hunk 2]
```diff
@@ -76,84 +71,41 @@ const Navigation: React.FC<NavigationProps> = ({navChildren}) => {
...
+              >
+                {item.label}
+              </Link>
+            )
+          })}
           {sellingLive && !isAuthenticated ? (
             <Link
               href="/login"
               className="rounded-lg px-3 py-2 text-text transition-all duration-150 ease-in-out lg:bg-blue-500 lg:text-white lg:hover:bg-blue-600"
+              data-navi-item=""
```
Comment (zacjones93) [inline apps/epic-react/src/components/app/navigation/index.tsx:104]: what's this attribute do?

Comment (zacjones93) [reply apps/epic-react/src/components/app/navigation/index.tsx:104]: Ah I see

https://github.com/skillrecordings/products/pull/1534/files#r1605615240

File: apps/epic-react/src/components/app/navigation/mobile-navigation.tsx (hunk 1 above)
Comment (zacjones93) [reply apps/epic-react/src/components/app/navigation/mobile-navigation.tsx:15]: So `naviItem` will be set if it's currently being rendered and undefined if not on screen. Haven't seen this approach before

**Review by zacjones93 — APPROVED** [review]

**Pull request merged**